### PR TITLE
Update Count_Min.mdx

### PR DESCRIPTION
**Pull request by dmiao623**
From [here](https://github.com/cpinitiative/usaco-guide/pull/2136)

### CODE REVIEW

**Review by Aadit-Ambadkar — APPROVED** [review]

File: content/6_Advanced/Count_Min.mdx [hunk 1]
```diff
@@ -3,38 +3,219 @@ id: count-min
...
+
+<CPPSection>
+
+```cpp
+const int MAXN = 2e5;
+const int INF = INT32_MAX;
+
+struct Node {
+	int val = INF, cnt = 1;
+} T[2 * MAXN];
```
Comment (SansPapyrus683) [inline content/6_Advanced/Count_Min.mdx:47]: rename T to something else

File: content/6_Advanced/Count_Min.mdx [hunk 2]
```diff
@@ -3,38 +3,219 @@ id: count-min
...
-**none** of the input rectangles.
+### Implementation
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+const int MAXN = 2e5;
+const int INF = INT32_MAX;
```
Comment (SansPapyrus683) [inline content/6_Advanced/Count_Min.mdx:43]: just use INT32_MAX what

File: content/6_Advanced/Count_Min.mdx [hunk 3]
```diff
@@ -3,38 +3,219 @@ id: count-min
...
+
+struct Node {
+	int val = INF, cnt = 1;
+} T[2 * MAXN];
+
+// combines two segment tree nodes
+Node merge(Node a, Node b) {
+	if (a.val < b.val)
+		return a;
```
Comment (SansPapyrus683) [inline content/6_Advanced/Count_Min.mdx:52]: add braces

File: content/6_Advanced/Count_Min.mdx [hunk 4]
```diff
@@ -3,38 +3,219 @@ id: count-min
...
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+const int MAXX = 1e6;
+const int MAXN = 2 * MAXX + 1;
```
Comment (SansPapyrus683) [inline content/6_Advanced/Count_Min.mdx:120]: i'd like it if you just defined MAXN as 2 mil
then you used MAXN + 1

File: content/6_Advanced/Count_Min.mdx [hunk 5]
```diff
@@ -3,38 +3,219 @@ id: count-min
...
+	// t = 1 for left bound, -1 for right bound
+	bool operator<(const Event &e) { return x < e.x; }
+};
+ 
+int N;
+vector<Event> E;
+	
+// segment tree
+pair<long long, long long> T[MAXN * 4];
+int L[MAXN * 4];
```
Comment (SansPapyrus683) [inline content/6_Advanced/Count_Min.mdx:133]: what the heck are T and L

File: content/6_Advanced/Count_Min.mdx [hunk 6]
```diff
@@ -3,38 +3,219 @@ id: count-min
...
+ 
+pair<long long, long long> merge(pair<long long, long long> a, pair<long long, long long> b) {
```
Comment (SansPapyrus683) [inline content/6_Advanced/Count_Min.mdx:135]: good job not using macros

File: content/6_Advanced/Count_Min.mdx [hunk 7]
```diff
@@ -3,38 +3,219 @@ id: count-min
...
+		T[t] = MP(0, 1);
+		return;
+	}
+	int tm = (tl + tr) >> 1;
+	build(t << 1, tl, tm);
+	build(t << 1 | 1, tm + 1, tr);
+	T[t] = merge(T[t << 1], T[t << 1 | 1]);
+}
+ 
+void update(int l, int r, int v, int t = 1, int tl = 0, int tr = MAXN) {
```
Comment (SansPapyrus683) [inline content/6_Advanced/Count_Min.mdx:164]: tell the user what these updates and queries actually mean maybe
(also pushdown what)

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
ok i looked at it again it hurts my eyes slightly less this time

File: content/6_Advanced/Count_Min.mdx (hunk 7 above)
Comment (dutinmeow) [reply content/6_Advanced/Count_Min.mdx:164]: these are regular lazy segtree function?

File: content/6_Advanced/Count_Min.mdx (hunk 1 above)
Comment (dutinmeow) [reply content/6_Advanced/Count_Min.mdx:47]: I think T is fine? It's what most people use

File: content/6_Advanced/Count_Min.mdx (hunk 7 above)
Comment (SansPapyrus683) [reply content/6_Advanced/Count_Min.mdx:172]: i have never coded a single pushdown in my life

File: content/6_Advanced/Count_Min.mdx (hunk 4 above)
Comment (dutinmeow) [reply content/6_Advanced/Count_Min.mdx:122]: This is much clearer, seeing as its one based and you have to multiply by 4 for segment trees.

Comment (dutinmeow) [reply content/6_Advanced/Count_Min.mdx:122]: I think maintaining a buffer of a couple of elements is a normal practice? Most people do `MAXN = 1e5 + 5` or smth

File: content/6_Advanced/Count_Min.mdx (hunk 7 above)
Comment (dutinmeow) [reply content/6_Advanced/Count_Min.mdx:171]: bruh

File: content/6_Advanced/Count_Min.mdx (hunk 4 above)
Comment (SansPapyrus683) [reply content/6_Advanced/Count_Min.mdx:122]: if you need a buffer for your code to run normally then something is wrong with your code
but i guess you're right

File: content/6_Advanced/Count_Min.mdx [hunk 8]
```diff
@@ -3,38 +3,227 @@ id: count-min
...
+	// t = 1 for left bound, -1 for right bound
+	bool operator<(const Event &e) { return x < e.x; }
+};
+ 
+int N;
+vector<Event> E;
+	
+// segment tree
+pair<long long, long long> Tree[MAXN * 4];
+int Lazy[MAXN * 4];
```
Comment (SansPapyrus683) [inline content/6_Advanced/Count_Min.mdx:135]: why in god's green earth are these variables capitalized

Comment (dutinmeow) [reply content/6_Advanced/Count_Min.mdx:135]: because they're global

Comment (SansPapyrus683) [reply content/6_Advanced/Count_Min.mdx:135]: what
what kind of convention do you follow

Comment (dutinmeow) [reply content/6_Advanced/Count_Min.mdx:135]: does it really matter as long as what the variables do is clear

Comment (SansPapyrus683) [reply content/6_Advanced/Count_Min.mdx:135]: honestly yes
if every single other solution (even benq's) variable names don't use this convention
you probably shouldn't yourself

Comment (dutinmeow) [reply content/6_Advanced/Count_Min.mdx:135]: Benq's variable names should be used as a counter example for good naming conventions

Comment (SansPapyrus683) [reply content/6_Advanced/Count_Min.mdx:135]: his macros are awful, yes
but at least his variable names are consistent
they use camelCase, or are just one letter
PascalCase is only for classes

**Comment by SansPapyrus683** [comment]
admins please don't kill me this has 4 approvals 2 of which are cm's i hope that's enough

**Pull request merged**